src/widgets/radio.jsx

```diff
@@ -579,17 +656,32 @@ var RadioEditor = React.createClass({
 
     onDelete: function(choiceIndex, e) {
         e.preventDefault();
+
         var choices = this.props.choices.slice();
+        var deleted = choices[choiceIndex];
+
         choices.splice(choiceIndex, 1);
-        this.props.onChange({choices: choices});
+
+        this.props.onChange({
+            choices: choices,
+            hasNoneOfTheAbove: this.props.hasNoneOfTheAbove && !deleted.isNoneOfTheAbove
+        });
     },
 
-    addChoice: function(e) {
+    addChoice: function(noneOfTheAbove, e) {
         e.preventDefault();
 
-        var choices = this.props.choices;
-        this.props.onChange({choices: choices.concat([{}])}, () => {
-            this.refs["editor" + choices.length].focus();
+        var choices = this.props.choices.slice();
+        var newChoice = { isNoneOfTheAbove: noneOfTheAbove };
+        var insertIndex = choices.length - ( this.props.hasNoneOfTheAbove || 0 );
```

Comment:
Following up on our previous discussion, if this is to turn a boolean into a number, could we change this to `? 1 : 0`? I think that makes it clearer what is happening.